src/domain/position/fx-timeout-reserved.js

```diff
@@ -0,0 +1,153 @@
+ *
+ * @async
+ * @description This is the domain function to process a bin of timeout-reserved messages of a single participant account.
+ *
+ * @param {array} fxTimeoutReservedBins - an array containing timeout-reserved action bins
+ * @param {number} accumulatedPositionValue - value of position accumulated so far from previous bin processing
+ * @param {number} accumulatedPositionReservedValue - value of position reserved accumulated so far, not used but kept for consistency
+ * @param {object} accumulatedFxTransferStates - object with commitRequest id keys and fxTransfer state id values. Used to check if fxTransfer is in correct state for processing. Clone and update states for output.
+ * @param {object} transferInfoList - object with transfer id keys and transfer info values. Used to pass transfer info to domain function.
+ * @returns {object} - Returns an object containing accumulatedPositionValue, accumulatedPositionReservedValue, accumulatedTransferStateChanges, accumulatedFxTransferStates, resultMessages, limitAlarms or throws an error if failed
+ */
+const processPositionFxTimeoutReservedBin = async (
+  fxTimeoutReservedBins,
+  accumulatedPositionValue,
+  accumulatedPositionReservedValue,
+  accumulatedFxTransferStates,
+  fetchedReservedPositionChangesByCommitRequestIds
+) => {
+  const fxTransferStateChanges = []
+  const participantPositionChanges = []
+  const resultMessages = []
+  const accumulatedFxTransferStatesCopy = Object.assign({}, accumulatedFxTransferStates)
+  let runningPosition = new MLNumber(accumulatedPositionValue)
+  // Position action FX_RESERVED_TIMEOUT event messages are keyed with payer account id.
+  // We need to revert the payer's position for the source currency amount of the fxTransfer.
+  // We need to notify the payee of the timeout.
+  if (fxTimeoutReservedBins && fxTimeoutReservedBins.length > 0) {
+    for (const binItem of fxTimeoutReservedBins) {
+      Logger.isDebugEnabled && Logger.debug(`processPositionFxTimeoutReservedBin::binItem: ${JSON.stringify(binItem.message.value)}`)
+      const participantAccountId = binItem.message.key.toString()
+      const commitRequestId = binItem.message.value.content.uriParams.id
+      const fxp = binItem.message.value.to
+      const payerFsp = binItem.message.value.from
+
+      // If the transfer is not in `RESERVED_TIMEOUT`, a position fx-timeout-reserved message was incorrectly published.
+      // i.e Something has gone extremely wrong.
+      if (accumulatedFxTransferStates[commitRequestId] !== Enum.Transfers.TransferInternalState.RESERVED_TIMEOUT) {
+        throw ErrorHandler.Factory.createInternalServerFSPIOPError(ErrorHandler.Enums.FSPIOPErrorCodes.INTERNAL_SERVER_ERROR.message)
+      } else {
+        Logger.isDebugEnabled && Logger.debug(`accumulatedFxTransferStates: ${JSON.stringify(accumulatedFxTransferStates)}`)
+
+        const transferAmount = fetchedReservedPositionChangesByCommitRequestIds[commitRequestId][participantAccountId].value
```

Comment:
As per my earlier comment, we need to make sure if we get single participantPositionChange item always. Otherwise we need to get the list of position changes and aggregate the resultant position value to be reverted.

Reply:
Makes sense, but do we ever forsee a operation doing a RESERVED transfer state change doing having multiple participant position changes?

Reply:
Yeah I can't see any use case atm. Lets add a comment illustrating that issue as per my other comment.